File: Java/Test00.java
import java.io.*;

public class Test00 {
  static HelloWorld student = new HelloWorld();
  private static ByteArrayOutputStream TOut;
  private static ByteArrayInputStream TIn;
  private static final PrintStream SOut = System.out;
  private static final InputStream SIn = System.in;
  public static String[] args;

  public static void main(String[] args){
    System.out.println(tests());
  }

  public static String tests(){
    int total = 0;
    int score = 0;
    setOutput();
    //test1
    total++;
    student.main(args);
    //need to add '/r' for windows when grading, stupid windows
    String correct = "Hello World!\n";
    String result = getOutput();
    if (result.equals(correct)){
      score++;
    }
    //no hidden tests for assignment 00
    //testing complete
    restoreSystem();
    String rep = ""+ score +"/"+total;
    return rep;
  }

  //Set up methods
  public static void setOutput(){
    TOut = new ByteArrayOutputStream();
    System.setOut(new PrintStream(TOut));
  }
  private static void setInput(String data){
  TIn = new ByteArrayInputStream(data.getBytes());
  System.setIn(TIn);
  }
  private static String getOutput(){
    return TOut.toString();
  }
  public static void restoreSystem(){
    System.setOut(SOut);
    System.setIn(SIn);
  }
}

File: Python/test_00.py
from subprocess import run, TimeoutExpired, Popen, PIPE
from os import getcwd
file = "hello_world.py"

# setup methods
def catchOutput(inputs=None, seed=''):
	cwd = getcwd()
	result = None
	p = Popen(f"python3 {file} {seed}",stdin=PIPE,stdout=PIPE,shell=True,cwd=cwd, text=True)
	try:
		out = p.communicate(input=inputs, timeout=3)
		result = out[0]
	except TimeoutExpired:
		result = ""
		p.kill()
		run("ps fjx > kill.txt ", shell=True)
		with open('kill.txt','r') as f:
			data = f.readlines()
		k = data[-1].split(" ")
		run(f"kill {k[5]}",shell=True)
	return result

def main():
	total = 0
	score = 0
	total += 1
	#def test_1():
	correct = "Hello World!\n"
	result = catchOutput()[:len(correct)]
	if result == correct:
		score += 1

	total += 1
	#def test_2():
	correct = "Hello World!\nNUAMES\n"
	result = catchOutput()[:len(correct)]
	if result == correct:
		score += 1

	total += 1
	#def test_3():
	correct = "Hello World!\nNUAMES\nCS 1030\n"
	result = catchOutput()
	if result == correct:
		score += 1
		
	#hidden tests
	#no hidden tests for this assignment

	return f"{score}/{total}"

if __name__ == "__main__":
	print(main())